### PR TITLE
added user edit page and functionality and added profile pic

### PR DESCRIPTION
**Pull request by sashaboi**
Edit user functionality added. please review that part.
Yet to do : styling.
auth decorator
redux 😭

### CODE REVIEW

File: src/pages/Homepage/homepage.css [hunk 1]
```diff
@@ -4,10 +4,12 @@
   justify-content: space-between;
 }
 .home-page-container {
-  margin: auto;
+  /* margin: auto;
```
Comment (bhavyaj12) [inline src/pages/Homepage/homepage.css:7]: Can remove commented css

File: src/pages/Profile/Profile.jsx [hunk 2]
```diff
@@ -1,5 +1,100 @@
...
-
+import axios from 'axios';
+import React, { useState } from 'react';
+import {
+  Footer,
+  Modal,
+  Navbar,
+  SideBarLeft,
+  SideBarRight,
+} from '../../components';
```
Comment (bhavyaj12) [inline src/pages/Profile/Profile.jsx:9]: It's better to use the convention of global imports first followed by local imports, makes code cleaner. So all react imports at the top, css at the last

**Review by bhavyaj12 — APPROVED** [review]
Code is easy to understand and well written.
Edit functionality works well. Minor improvements suggested, and of course styling is pending. I think you should add redux before any more functionality else it'll be hard to edit out everything 
Great work!

File: src/pages/Profile/profile.css [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+.edit-form {
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  gap: 2rem;
+  padding: 2rem;
+}
+.form-element {
+  display: flex;
+  flex-direction: column;
```
Comment (SHIKHASINGH1506) [inline src/pages/Profile/profile.css:17]: I can see this style is being used multiple times hence making your code redundant. You can create a small utility for flex that will help you at many places as you're heavily using it

File: src/components/PostCard/PostCard.jsx [hunk 4]
```diff
@@ -110,21 +111,29 @@ export const PostCard = ({ post }) => {
     <div className="post-card-parent">
       <div className="post-top-content">
         <div className="profile-data">
-          <img src="" alt="profile pic" />
+          <div className="profile-pic-letters">
+            {userOfPost[0].firstName.slice(0, 1)}
+            {userOfPost[0].lastName.slice(0, 1)}
```
Comment (SHIKHASINGH1506) [inline src/components/PostCard/PostCard.jsx:116]: I'm not getting use of slice here

**Review by SHIKHASINGH1506 — APPROVED** [review]
1. CSS are repetitive at few places depending on your bandwidth you can refactor it
2. Other than that code looks fine

File: src/components/PostCard/PostCard.jsx (hunk 4 above)
Comment (sashaboi) [reply src/components/PostCard/PostCard.jsx:116]: I am making the profile picture from the first letter of firstname and lastname , hence the slice.

File: src/pages/Profile/profile.css (hunk 3 above)
Comment (sashaboi) [reply src/pages/Profile/profile.css:17]: I agree, will refactor this in coming versions.

**Pull request merged**